### PR TITLE
community[minor]: DeepInfra embeddings integration #1

### PR DESCRIPTION
**Pull request by ovuruska**
**Description**

Implemented a new embedding integration that makes DeepInfra's embedding models available on langchainjs. A new class named DeepInfraEmbeddings is created, which wraps the inference API.

This PR includes an example and integration tests related to the newly implemented class. Apart from the new integration's configuration, the original open-source components are intentionally left unchanged to avoid introducing any breaking changes.

### CODE REVIEW

File: libs/langchain-community/src/embeddings/deepinfra.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
+import axios, {AxiosInstance, AxiosResponse} from "axios";
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/deepinfra.ts:1]: `axios` isn't a dependency - can you use plain `fetch` instead?

Comment (jacoblee93) [reply libs/langchain-community/src/embeddings/deepinfra.ts:1]: Please also run `yarn format`

File: libs/langchain-community/src/embeddings/deepinfra.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+  /**
+   * Generates embeddings for an array of texts.
+   * @param inputs - An array of strings to generate embeddings for.
+   * @returns A Promise that resolves to an array of embeddings.
+   */
+  async embedDocuments(inputs: string[]): Promise<number[][]> {
+    await this.maybeInitClient();
+
+    const batches = chunkArray(inputs, this.batchSize) as string[][];
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/deepinfra.ts:138]: Cast shouldn't be necessary?

File: libs/langchain-community/src/embeddings/deepinfra.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  /**
+   * Generates embeddings with retry capabilities.
+   * @param request - An object containing the request parameters for generating embeddings.
+   * @returns A Promise that resolves to the API response.
+   */
+  private async embeddingWithRetry(
+    request: DeepInfraEmbeddingsRequest
+  ): Promise<DeepInfraEmbeddingsResponse> {
+    this.maybeInitClient();
+    const response = await this.caller.call(this.client.post.bind(this.client,""), request);
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/deepinfra.ts:184]: What is the second arg you're binding here?

File: libs/langchain-community/src/embeddings/deepinfra.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        headers: {
+          Authorization: `Bearer ${this.apiToken}`,
+          ContentType: "application/json",
+        },
+      });
+    }
+  }
+
+  /** @ignore */
+  static async imports(): Promise<{}> {
```
Comment (jacoblee93) [inline libs/langchain-community/src/embeddings/deepinfra.ts:205]: Remove

**Review by jacoblee93 — CHANGES_REQUESTED** [review]

**Comment by jacoblee93** [comment]
Thank you, sorry for the delay!

**Comment by ovuruska** [comment]
@jacoblee93 Thanks for the feedback!

**Comment by jacoblee93** [comment]
Looks good, can you add a documentation page under `docs/core_docs/docs/integrations/text_embedding/` to accompany the examples though?

**Comment by ovuruska** [comment]
Thanks @jacoblee93 , I updated the documentation.

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**